### ISSUE TITLE
Test coverage integration problems, espec and Elixir ~> 1.1.0

### ISSUE THREAD

**Issue by dm1try** [antonmi/espec#64]
I've noticed that  CI builds with Elixir 1.1.1 and `espec` do not generate a coverage report.
It does not depend of a coverage tool(standard/coverex/etc) though. ~> 1.0.5 does not have such problem.

If I use a standard coverage tool and some sample project:
```
MIX_ENV=test mix espec --cover
Cover compiling modules ... 
** (Mix) Failed to cover compile directory: /home/dmitry/coverage_sample/_build/test/lib/coverage_sample/ebin
```
But with pre-compilation it works fine
```
 MIX_ENV=test mix compile      
 MIX_ENV=test mix espec --cover
...
Generating cover results ...
```
Seems [here](https://github.com/antonmi/espec/blob/master/lib/mix/tasks/espec.ex#L104-L111) we should ensure that an app already compiled?

If we add `Mix.Task.run "compile.app"` before the starting a cover tool this resolves the problem.
Also `Mix.Project.compile(project)` will work but only for Elixir ~> 1.1.0

Thoughts?

**Comment by antonmi** [comment]
Hi, Dmitry!
I tried to reproduce the issue, but I couldn't.
Whould you share the code of your cover_sample application?
As I understand Mix tool might do compilation before running mix task. So there is no need to explicitly "compile.app".
For now, just help me to reproduce the issue.
Thanks!

**Comment by dm1try** [comment]
Hi, Anton!
It's a freshly-generated app (ex. `mix new cov_test`). I just installed `espec` (added to deps and run `deps.get` + `MIX_ENV=test mix espec.init`)

**BUT** make sure you run `mix clean` after `MIX_ENV mix espec.init` and before `MIX_ENV mix espec --cover`:
```
mix clean
MIX_ENV=test mix espec --cover
Cover compiling modules ... 
** (Mix) Failed to cover compile directory: /home/dmitry/Work/Projects/cov_test/_build/test/lib/cov_test/ebin

mix clean
MIX_ENV=test mix test --cover 
Compiled lib/cov_test.ex
Generated cov_test app
Cover compiling modules ... 
.

Finished in 0.05 seconds (0.05s on load, 0.00s on tests)
1 test, 0 failures

Randomized with seed 426983

Generating cover results ... 
```
If it won't help you can try https://github.com/dm1try/jequalson :) (here I use `coverex` but it's the same problem). You can see the build logs [Elixir 1.0.5](https://travis-ci.org/dm1try/jequalson/jobs/91198975), [Elixir 1.1.1](https://travis-ci.org/dm1try/jequalson/jobs/91198976). I enabled the :debug level for the `coverex` so the results a little bit messy.

**Comment by antonmi** [comment]
I've reproduced.
I've added 'if Keyword.get(opts, :compile, true), do: Mix.Task.run("compile", args)' to mix task.
And released 0.8.6 with this and some other fixes.
Thanks!

**Comment by dm1try** [comment]
@antonmi thanks for the release  :+1:  it works like a charm.
